include/acquire.zarr.h

```diff
@@ -0,0 +1,112 @@
+#pragma once
+
+#include "zarr.types.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+    /**
+     * @brief The settings for a Zarr stream.
+     * @details This struct contains the settings for a Zarr stream, including
+     * the store path, custom metadata, S3 settings, chunk compression settings,
+     * dimension properties, whether to stream to multiple levels of detail, the
+     * pixel data type, and the Zarr format version.
+     * @note The store path can be a filesystem path or an S3 key prefix. For example,
+     * supplying an endpoint "s3://my-endpoint.com" and a bucket "my-bucket" with a
+     * store_path of "my-dataset.zarr" will result in the store being written to
+     * "s3://my-endpoint.com/my-bucket/my-dataset.zarr".
+     * @note The dimensions array may be allocated with ZarrStreamSettings_create_dimension_array
+     * and freed with ZarrStreamSettings_destroy_dimension_array. The order in which you
+     * set the dimension properties in the array should match the order of the dimensions
+     * from slowest to fastest changing, for example, [Z, Y, X] for a 3D dataset.
+     */
+    typedef struct ZarrStreamSettings_s
+    {
+        const char* store_path; /**< Path to the store. Filesystem path or S3 key prefix. */
+        const char* custom_metadata; /**< JSON-formatted custom metadata to be stored with the dataset. */
+        ZarrS3Settings* s3_settings; /**< Optional S3 settings for the store. */
+        ZarrCompressionSettings* compression_settings; /**< Optional chunk compression settings for the store. */
+        ZarrDimensionProperties* dimensions; /**< The properties of each dimension in the dataset. */
+        size_t dimension_count; /**< The number of dimensions in the dataset. */
+        bool multiscale; /**< Whether to stream to multiple levels of detail. */
+        ZarrDataType data_type; /**< The pixel data type of the dataset. */
+        ZarrVersion version; /**< The version of the Zarr format to use. 2 or 3. */
+    } ZarrStreamSettings;
+
+    typedef struct ZarrStream_s ZarrStream;
+
+    /**
+     * @brief Get the version of the Zarr API.
+     * @return The version of the Zarr API.
+     */
+    uint32_t Zarr_get_api_version();
+
+    /**
+     * @brief Set the log level for the Zarr API.
+     * @param level The log level.
+     * @return ZarrStatusCode_Success on success, or an error code on failure.
+     */
+    ZarrStatusCode Zarr_set_log_level(ZarrLogLevel level);
```

Comment:
I left questions about the log API that were not answered in the previous pass. While not critical, I believe there *might* be potential for improvement in this area.